### PR TITLE
[MAPSME-2514] [ios] Updated Welcome/What’s New screens.

### PR DESCRIPTION
**Pull request by igrechuhin**
https://jira.mail.ru/browse/MAPSME-2514

### CODE REVIEW

File: iphone/Maps/UI/Welcome/FirstLaunchController.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import UIKit
+
+final class FirstLaunchController: MWMViewController, WelcomeController {
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/FirstLaunchController.swift]: s/WelcomeController/WelcomeProtocol/g

File: iphone/Maps/UI/Welcome/WelcomeController.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  weak var nextPageButton: UIButton! { get set }
+  weak var containerWidth: NSLayoutConstraint! { get set }
+  weak var containerHeight: NSLayoutConstraint! { get set }
+  weak var imageMinHeight: NSLayoutConstraint! { get set }
+  weak var imageHeight: NSLayoutConstraint! { get set }
+  weak var titleTopOffset: NSLayoutConstraint! { get set }
+  weak var titleImageOffset: NSLayoutConstraint! { get set }
+}
+
+protocol WelcomeController: WelcomeControllerBase {
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/WelcomeController.swift]: move below

File: iphone/Maps/UI/Welcome/WelcomeController.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
+protocol WelcomeControllerBase: class {
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/WelcomeController.swift]: s/WelcomeControllerBase/WelcomeProtocolBase/g

File: iphone/Maps/UI/Welcome/WelcomePageController.swift [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    Statistics.logEvent(kStatEventName(kStatWhatsNew, welcomeKey),
+                        withParameters: [kStatAction : kStatOpen])
+    parentController.addChildViewController(self)
+    parentController.view.addSubview(view)
+    updateFrame()
+  }
+
+  private func updateFrame() {
+    let parentView = parentController.view!
+    let size: CGSize
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: This is CGRect, isn't it?

File: iphone/Maps/UI/Welcome/WelcomePageController.swift [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+extension WelcomePageController: UIPageViewControllerDataSource {
+
+  func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+    let index = controllers.index(of: viewController)!
+    return index == 0 ? nil : controllers[index - 1]
+  }
+
+  func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+    let index = controllers.index(of: viewController)!
+    return index == controllers.count - 1 ? nil : controllers[index + 1]
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: Maybe 
```swift
return index == controllers.last == viewController ? nil : controllers[index + 1]
```

File: iphone/Maps/UI/Welcome/WelcomePageController.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  }
+
+  func presentationCount(for pageViewController: UIPageViewController) -> Int {
+    return controllers.count
+  }
+
+  func presentationIndex(for pageViewController: UIPageViewController) -> Int {
+    guard let vc = viewControllers?.first else { return 0 }
```
Comment (syershov) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: currentController?

File: iphone/Maps/UI/Welcome/FirstLaunchController.swift (hunk 1 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/FirstLaunchController.swift]: OK

File: iphone/Maps/UI/Welcome/FirstLaunchController.swift [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+  typealias ConfigBlock = (FirstLaunchController) -> ()
+  static var pagesConfigBlocks: [ConfigBlock]! {
+    return [
```
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/FirstLaunchController.swift]: Fixed

File: iphone/Maps/UI/Welcome/WelcomeController.swift (hunk 2 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/WelcomeController.swift]: OK

File: iphone/Maps/UI/Welcome/WelcomeController.swift (hunk 3 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/WelcomeController.swift]: OK

File: iphone/Maps/UI/Welcome/WelcomePageController.swift (hunk 4 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: OK

File: iphone/Maps/UI/Welcome/WelcomePageController.swift (hunk 5 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: OK

File: iphone/Maps/UI/Welcome/WelcomePageController.swift (hunk 6 above)
Comment (igrechuhin) [inline iphone/Maps/UI/Welcome/WelcomePageController.swift]: OK

**Pull request merged**